<?xml version="1.0"?>
<robot name="front_robot"
  xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:property name="M_PI" value="3.1415926"/>


  <!-- base_link -->
  <link name="base_link">

    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.5 0.5 0.25"/>
      </geometry>
      <material name="blue">
        <color rgba="0.0 0.5 1.0 1.0"/>
      </material>
    </visual>

    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.5 0.5 0.25"/>
      </geometry>
    </collision>

    <inertial>
      <mass value="5"/>
      <inertia ixx="0.1" ixy="0.0" ixz="0.0" iyy="0.1" iyz="0.0" izz="0.1"/>
    </inertial>

  </link>

  <gazebo reference="base_link">
    <material>Gazebo/Blue</material>
  </gazebo>

  <!-- right wheel -->
  <link name="right_wheel">

    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="${M_PI/2} 0.0 0.0"/>
      <geometry>
        <cylinder radius="0.2" length="0.1"/>
      </geometry>
      <material name="black">
        <color rgba="0.05 0.05 0.05 1.0"/>
      </material>
    </visual>

    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="${M_PI/2} 0.0 0.0"/>
      <geometry>
        <cylinder radius="0.2" length="0.1"/>
      </geometry>
    </collision>

    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.005" iyz="0.0" izz="0.005"/>
    </inertial>

  </link>

  <gazebo reference="right_wheel">
    <material>Gazebo/Black</material>
  </gazebo>


  <!-- left wheel -->
  <link name="left_wheel">
    <visual>
      <material name="black"/>
      <origin xyz="0.0 0.0 0.0" rpy="${M_PI/2} 0.0 0.0"/>
      <geometry>
        <cylinder radius="0.2" length="0.1"/>
      </geometry>
    </visual>

    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="${M_PI/2} 0.0 0.0"/>
      <geometry>
        <cylinder radius="0.2" length="0.1"/>
      </geometry>
    </collision>

    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.005" iyz="0.0" izz="0.005"/>
    </inertial>

  </link>

  <gazebo reference="left_wheel">
    <material>Gazebo/Black</material>
  </gazebo>

  <!-- joint right -->
  <joint name="joint_right_wheel" type="continuous">
    <origin xyz="0.0 -0.3 0.0" rpy="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="right_wheel"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>

  <!-- joint left -->
  <joint name="joint_left_wheel" type="continuous">
    <origin xyz="0.0 0.3 0.0" rpy="0.0 0.0 0.0"/>
    <parent link="base_link"/>
    <child link="left_wheel"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>


</robot>